tools/local-dev.sh

```diff
@@ -6,27 +6,42 @@ set -e
 ## Usage: source tools/local-dev.sh
 
 ## The script assumes that it is being run from the top-level directory "terra-cli/".
-if [ $(basename $PWD) != 'terra-cli' ]; then
+if [ "$(basename "$PWD")" != 'terra-cli' ]; then
   echo "Script must be run from top-level directory 'terra-cli/'"
   exit 1
 fi
 
+echo "Determining installation mode"
+if [ -z "$TERRA_CLI_INSTALLATION_MODE" ] || [ "$TERRA_CLI_INSTALLATION_MODE" == "SUPPORT_DOCKER" ]; then
+  terraCliInstallationMode="SUPPORT_DOCKER"
+elif [ "$TERRA_CLI_INSTALLATION_MODE" == "LOCAL_ONLY" ]; then
+  terraCliInstallationMode="LOCAL_ONLY"
+else
+  echo "Unsupported TERRA_CLI_INSTALLATION_MODE specified: $TERRA_CLI_INSTALLATION_MODE"
+  exit 1
+fi
+
+echo "Installation mode is $terraCliInstallationMode"
+
 echo "Building Java code"
 ./gradlew clean install
 
 echo "Aliasing JAR file"
-alias terra=$(pwd)/build/install/terra-cli/bin/terra
-
-echo "Setting the Docker image id to the default"
-terra config set image --default
-
-echo "Pulling the default Docker image"
-defaultDockerImage=$(terra config get image)
-docker pull $defaultDockerImage
+alias terra="$(pwd)"/build/install/terra-cli/bin/terra
+
+if [ "$terraCliInstallationMode" == "LOCAL_ONLY" ]; then
+  echo "Installing without docker support"
+else
+  echo "Setting the Docker image id to the default"
+  terra config set image --default
```

Comment:
There's a related issue of whether we should allow setting the image before the image has been pulled (or if it otherwise isn't spelled right, etc). I had put in code to bail if it wasn't there, but this broke `install-for-testing.sh` and likely `install.sh` too.

For now I'm going to remove the check & throw I had put in `bio.terra.cli.businessobject.Config#setDockerImageId` and revisit under [PF-1332](https://broadworkbench.atlassian.net/browse/PF-1332).

Reply:
Right now, we allow setting the image id regardless of whether the image is there or not. This is particularly helpful during install, because then we can set the image id to the default and then ask the cli what the default is. So the install script doesn't need to know what the default is explicitly.

I think it would be good to set the image id to the default regardless of what the app-launch config property is, because the user may toggle that property later, and if so, we want them to pull the latest cli image.